### PR TITLE
Edit idea via ajax

### PR DESCRIPTION
**Pull request by jeffreyfultonca**
Ability to edit an Idea's title and description inline with ajax.
Moved upvote JavaScript into compiled CoffeeScript asset.

### CODE REVIEW

File: app/assets/javascripts/ideas.js.coffee [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		title = list_item_form.find('#idea_title').val()
+		description = list_item_form.find('#idea_description').val()
+
+		list_item_labels.find('.title').text(title)
+		list_item_labels.find('.description').text(description)
+
+		list_item_labels.toggle('slow')
+		list_item_form.toggle('slow')
+
+	$(".upvote img").click ->
```
Comment (swalberg) [inline app/assets/javascripts/ideas.js.coffee]: Thank you!

File: app/assets/javascripts/ideas.js.coffee [hunk 2]
```diff
@@ -0,0 +1,29 @@
+$ ->
+	$('.edit').click ->
```
Comment (swalberg) [inline app/assets/javascripts/ideas.js.coffee]: Style - suggest 2 spaces for tab like in Ruby

File: spec/controllers/ideas_controller_spec.rb [hunk 3]
```diff
@@ -105,4 +105,30 @@
     end
   end
   
+  # describe '#edit' do
```
Comment (swalberg) [inline spec/controllers/ideas_controller_spec.rb]: You've excluded the :edit action in the routes so please drop the test.

File: spec/controllers/ideas_controller_spec.rb [hunk 4]
```diff
@@ -105,4 +105,30 @@
...
+      @attr = { title:'Updated title', description:'Updated description' }
+      put :update, id: @idea.id, idea: @attr
+
+    end
+
+    it { should respond_with :redirect }
+    it { should redirect_to :root }
+    
+    # How to test it actually got updated?
+    #it { assigns(:ideas).should == @ideas }
```
Comment (swalberg) [inline spec/controllers/ideas_controller_spec.rb]: In general, try to [avoid instance variables](http://betterspecs.org/#let) in rspec tests. Suggesting something like

``` Ruby
let(:ideas) { FactoryGirl.create_list :idea, 10 }
let(:idea) { idea.first }
let(:attrs) { { title:'Updated title', description:'Updated description' } }

before { put :update, id: idea.id, idea: attrs }

it "updates the idea" do
  expect(idea.reload.title).to eq attrs[:title]
end
```

Also, for future thought, try to prefer `expect` over `should` (http://betterspecs.org/#expect). Frankly I like the way you did it better, but that's the way rspec is moving.

**Comment by swalberg** [comment]
All my comments are stylistic in nature, so feel free to ignore them.

Normally I'd squash the 4 commits to 1, but again, at your discretion.

:+1:

**Comment by jeffreyfultonca** [comment]
The feedback is great. Much appreciated. Keep it up.

**Comment by jeffreyfultonca** [comment]
I assume I can just make changes, then re-submit the pull request?

**Comment by swalberg** [comment]
Commit to your local branch, do the `git push origin edit_idea_via_ajax`, and the existing PR will be updated automagically. Just think of this code review as a comment thread on a branch.

File: spec/controllers/ideas_controller_spec.rb (hunk 4 above)
Comment (swalberg) [inline spec/controllers/ideas_controller_spec.rb]: Looking at what I typed again, it should be `let(:idea) { ideas.first }` (plural inside the block)

**Comment by amirci** [comment]
I grabbed your commits and applied them after master, did a few changes to match the new UI, etc.

Please look at my branch f21_edit_idea_using_ajax and go from there to finish the feature.

I'll close the request for now.

**Comment by amirci** [comment]
I'll reopen the PR hoping that somebody will fix what's missing.

**Comment by jeffreyfultonca** [comment]
I'd like to take this.

Amir: Can you help me identify the procedure for completing this work regarding git? 

I have cloned your f21_edit_idea_using_ajax branch and am looking it over. Should I merge this into my local edit_idea_using_ajax branch? Add your branch as a remote and merge from there? Should rebase be involved?

**Comment by amirci** [comment]
I would use my branch as a start and go from there (it has all your changes).

```
git checkout -b f21_edit_idea_using_ajax amir/f21_edit_idea_using _ajax
```

This way you branch starting from my branch (tracking). Keep your branch with a different name just for reference.

**Comment by amirci** [comment]
Close by mistake

**Comment by amirci** [comment]
Close for now until it is finished

**Pull request closed without merging**